src/test/java/edu/ucsb/cs156/happiercows/controllers/CowDeathControllerTests.java

```diff
@@ -0,0 +1,277 @@
+    .cowsKilled(9)
+    .userId(5)
+    .build();
+  CowDeath cowdeathsample2 = CowDeath.builder()
+    .id(1)
+    .avgHealth(12)
+    .commonsId(1)
+    .cowsKilled(5)
+    .userId(14)
+    .build();
+
+
+  expectedCowDeaths.add(cowdeathsample);
+  expectedCowDeaths.add(cowdeathsample2);
+  when(cowdeathRepository.getCowsKilledByCommonsId(1L)).thenReturn(expectedCowDeaths);
+  when(cowdeathRepository.save(cowdeathsample)).thenReturn(cowdeathsample);
+  when(cowdeathRepository.save(cowdeathsample2)).thenReturn(cowdeathsample2);
+
+  MvcResult response = mockMvc.perform(get("/api/cowdeath/bycommons?commonsId=1").with(csrf()))
+            .andExpect(status().isOk()).andReturn();
+
+  verify(cowdeathRepository, times(1)).getCowsKilledByCommonsId(1L);
+
+  String responseString = response.getResponse().getContentAsString();
+  List<CowDeath> actualCowDeaths = objectMapper.readValue(responseString, new TypeReference<List<CowDeath>>() {});
+  assertEquals(expectedCowDeaths, actualCowDeaths);
+}
+
+@WithMockUser(roles = { "USER" })
+@Test
+public void get_cowdeaths_using_commons_id_and_user_id() throws Exception {
+  CowDeath cowDeathSample = CowDeath.builder()
+    .id(0)
+    .avgHealth(11)
+    .commonsId(1)
+    .cowsKilled(9)
+    .userId(100)
+    .build();
+
+
+  when(cowdeathRepository.getCowsKilledByCommonsIdAndUserId(1L, 100L)).thenReturn(Optional.of(cowDeathSample));
+
+  
+
+  MvcResult response = mockMvc.perform(get("/api/cowdeath/byusercommons?commonsId=1&userId=100").with(csrf()))
+            .andExpect(status().isOk()).andReturn();
+  verify(cowdeathRepository, times(1)).getCowsKilledByCommonsIdAndUserId(1L,100L);
+
+  String responseString = response.getResponse().getContentAsString();
+  CowDeath actualCowDeath = objectMapper.readValue(responseString, new TypeReference<CowDeath>() {});
+  assertEquals(cowDeathSample, actualCowDeath);
+}
+
+
+@WithMockUser(roles = { "USER" })
+@Test
+public void get_cowdeath__nonexistent_using_commons_id_and_user_id() throws Exception {
+  MvcResult response = mockMvc.perform(get("/api/cowdeath/byusercommons?commonsId=100000&userId=1").contentType("application/json")).andExpect(status().isNotFound()).andReturn();
+
+  verify(cowdeathRepository, times(1)).getCowsKilledByCommonsIdAndUserId(100000L,1L);
+
+  Map<String, Object> json = responseToJson(response);
+  assertEquals("EntityNotFoundException", json.get("type"));
+  assertEquals("CowDeath with commonsId 100000 and userId 1 not found", json.get("message"));
+}
+
+@WithMockUser(roles = { "ADMIN" })
+@Test
+public void get_cowdeath_nonexistent_using_commons_id() throws Exception {
+
+  MvcResult response = mockMvc.perform(get("/api/cowdeath/bycommons?commonsId=20000").contentType("application/json")).andExpect(status().isOk()).andReturn();
+
+  verify(cowdeathRepository, times(1)).getCowsKilledByCommonsId(20000L);
+
+  String responseString = response.getResponse().getContentAsString();
+  assertEquals("[]", responseString);
+}
+  
+}
```

Comment:
The indentation here is not standard.   Please follow the indentation conventions of the rest of the code base.

This is easy to do if you use VSCode and install the recommended Microsoft Java Extensions.

You just use the "format code" key stroke and it's all done for you.  It is  unprofessional to submit code for review that doesn't have consistent indentation.